grommet/grommet
# Filter out 'size' prop from leaking through to the DOM in <Text>

Fixes #4003

#### What does this PR do?
Filter out 'size' prop from leaking through to the DOM in <Text>

#### Where should the reviewer start?
StyledText.js

#### What testing has been done on this PR?
Run JEST tests and storybook

#### How should this be manually tested?
Use storybook to check that components which use <Text> look correct and have the correct text size

#### Any background context you want to provide?
&lt;Text&gt; passes 'size' to StyledText which uses 'styled' to convert it into CSS for the font. However, styled-components silently passes any props which are valid HTML props to the underlying target element. Since size is a valid HTML prop for inputs and selects, it gets passed onto the target element (no matter what the tag actually is.)

styled-components 5.1 introduced a config option for 'styled' which lets you pass a shouldForwardProp function to filter out props you don't want to make it to the element. Grommet had a peer dependency on styled-components >= 5.X so this was upgraded to 5.1
 
#### What are the relevant issues?
#4003 

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Should be backwards compatible
